### PR TITLE
Convert ints to doubles where necessary

### PR DESCRIPTION
**Pull request by hinescd**
Ints need to be wrapped with ```Double.valueOf(...)``` under some circumstances, like testing a double for equality with an integer (```4 == 4.0```)

This change makes tests/failing/add.lj work as expected, and now the only flaw with tests/failing/primitive-numbers.lj is floating point error which I don't think we can fix anyway.

### CODE REVIEW

File: src/main/java/com/github/lessjava/types/ast/ASTConvertIntToDouble.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+import com.github.lessjava.types.inference.impl.HMTypeBase;
+
+/**
+ * Converts Integer types to Double types by wrapping with Double.valueOf(...).
+ *
+ * Necessary because Java can't natively do this, so things like 4 == 4.0 would break unless the 4
+ * is converted to a Double.
+ */
+public class ASTConvertIntToDouble extends ASTExpression {
```
Comment (hinescd) [inline src/main/java/com/github/lessjava/types/ast/ASTConvertIntToDouble.java]: I'm open to renaming this as well as LJConvertIntsToDoubles if anyone has any better name ideas

**Comment by hinescd** [comment]
It doesn't look like we were already doing coercion, since these tests were failing before. I think this is an important feature to have, since new programmers would likely want to be able to have something like ```return 5``` inside a function that has a ```double``` return type.

I experimented with just changing the type attribute to ```HMTypeBase.REAL``` before this solution, but that didn't work because without this the compiler only generates the whole ```Double.valueOf(...)``` deal for literals, so if ```foo()``` returns a double 5.0, ```bar()``` returns integer 5, and we do ```test foo() == bar()```, the test will fail because the call to bar is not coerced to a double to match foo. This change allows us to wrap non-literals with ```Double.valueOf(...)```, so the test will pass now.

**Comment by lam2mo** [comment]
Actually, I think I recall that we decided that we **didn't** actually want coercions to preserve the strong typing. I think we eventually wanted conversion routines (e.g., `return double(5)`) but I don't recall specifics. Let's hold on this until we can talk about it tomorrow.

**Comment by hinescd** [comment]
Closed until #69 is resolved

**Pull request closed without merging**